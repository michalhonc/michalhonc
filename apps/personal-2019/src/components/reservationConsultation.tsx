import React from "react";
import { useIntl } from "gatsby-plugin-intl";

const CALENDLY_URL =
  "https://calendly.com/michalhonc/free?primary_color=04776a";

export const ReservationConsultation = () => {
  const intl = useIntl();

  return (
    <button
      type="button"
      className="ReservationConsultation"
      onClick={() => {
        if ((window as any).Calendly) {
          (window as any).Calendly.initPopupWidget({ url: CALENDLY_URL });
        } else {
          window.location.href = CALENDLY_URL;
        }
      }}
    >
      {intl.formatMessage({ id: "reservationConsultation" })}
    </button>
  );
};
