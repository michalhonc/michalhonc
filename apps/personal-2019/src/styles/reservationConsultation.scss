.ReservationConsultation {
  font-size: 1.8rem;
  color: #fff;
  padding: 1.5rem 3rem;
  background-color: #3D3F4C;
  border-radius: 8rem;
  box-shadow: 0 10px 10px -10px rgba(0, 0, 0, 0.5);
	transition: all 0.2s ease-out;
	transform: translateY(0);
	outline: none;

  @include respond('phone') {
    font-size: 1.4rem;
    padding: 1rem 2rem;
    max-width: 20rem;
  }

	&:focus,
	&:hover {
		transform: translateY(-0.2rem);
		box-shadow: 0 10px 20px -10px rgba(0, 0, 0, 0.5);
	}

	&:active {
		transform: translateY(-0.1rem);
	}
}
