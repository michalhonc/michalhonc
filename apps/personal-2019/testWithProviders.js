import React from "react";
import renderer from "react-test-renderer";
import { IntlContextProvider } from "gatsby-plugin-intl";

export const renderWithProviders = (component) =>
  renderer.create(
    <IntlContextProvider value={{ language: "en-US", routed: true }}>
      {component}
    </IntlContextProvider>
  );
