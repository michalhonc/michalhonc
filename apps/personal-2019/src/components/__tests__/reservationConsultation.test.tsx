import * as React from "react";
import renderer from "react-test-renderer";
import { renderWithProviders } from "../../../testWithProviders";
import { ReservationConsultation } from "../reservationConsultation";

describe("reservationConsultation", () => {
  it("renders correctly", () => {
    const tree = renderWithProviders(<ReservationConsultation />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
