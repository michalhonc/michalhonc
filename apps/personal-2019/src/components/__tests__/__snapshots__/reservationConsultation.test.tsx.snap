// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reservationConsultation renders correctly 1`] = `
<button
  className="ReservationConsultation"
  onClick={[Function]}
  type="button"
>
  Rezervovat konzultaci zdarma
</button>
`;
